src/ape/managers/project/types.py

```diff
@@ -170,7 +173,7 @@ def _as_ape_project(self, **kwargs):
 
     def create_manifest(
         self, file_paths: Optional[List[Path]] = None, use_cache: bool = True
-    ) -> PackageManifest:
+    ) -> Tuple[PackageManifest, Dict[str, ContractType]]:
```

Comment:
`contract_types` is a property on the `PackageManifest`, so could we set the contract types property on the manifest and only return the manifest?

Reply:
Can we make the cached manifest also grab the type JSONs? Like can we have it be excluded from the cached file but not excluding from the Python object?

Thinking out-loud here. I really appreciate you working on this.

Reply:
This might look like a local project implementing this method only after all the other files are created and are consistent with each other (as a result of an intermediate pass we use for compiling a local project, likely many times in a row), compiling the manifest to extract only when necessary for the publishing step